File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        x = find(nums , 0 , len(nums)-1,target)
        return x

def find(nums , start , end , target):
    
    if end >= start:
        
        mid = (start + end)//2

        if nums[mid] == target:
            return mid

        if target > nums[mid]:
            start = mid + 1
            x = find(nums , start , end , target)
            return x
        else:
            end = mid - 1
            x = find(nums , start , end , target)
            return x
    else:
        return -1
    
    
        

File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int x = bs(nums , target , 0 , nums.length-1);
        return x;
    }
    
    public int bs(int[] nums , int target , int s , int e){
        if(s > e){
            return -1;
        }
        
        int mid = (s+e)/2;
        if(nums[mid] == target){
            return mid;
        }
        if(nums[mid] > target){
            return bs(nums , target ,s , mid-1 );
        }
        
        return bs(nums , target , mid+1 , e);
    }
}